crates/ironrdp-cliprdr/src/pdu/format_list.rs

```diff
@@ -13,6 +13,15 @@ pub struct ClipboardFormat {
     pub name: String,
 }
 
+impl ClipboardFormat {
+    pub fn new_standard(id: u32) -> Self {
+        Self {
+            id,
+            name: String::new(),
```

Comment:
In what cases is `name` empty? Maybe this type should be `Optional<String>` instead

Reply:
AFIK It is non-empty for os-specific formats, so it could be used for mapping remote and local clipboard IDs between each other.

I am not sure that we need `Option` here. because CLIPRDR specification tells us that when the name is not specified, it should be represented as the single null terminator, which translates to an empty string, Idk if it will make an API cleaner, @CBenoit what is your opinion on that?

Reply:
It’s generally more idiomatic to distinguish between absence of value, and the empty string in Rust, so here are the questions:
- Do we actually need to distinguish the empty string from the "no string" state?
- How ignoring this distinction could be problematic? What kind of issues the consumer of the API may run into?

In this case, what is the meaning of the empty string in the context of `ClipboardFormat`? I believe it means "non standard" or "OS-specific" format. So my understanding is that the empty value is actually something different than _just_ an empty string. It’s likely that special handling is required in such situations.

Another way to see it: the consumer of the API may not even be aware that the empty string requires any special handling and may miss this detail, or may not know how to initialize the value properly. Could the user end up asking himself "I don’t know what I’m supposed to put inside this string"? It’s likely that `Option` will make it obvious that `None` is a valid avenue.

Not only `Option` would be a valid avenue, sum types in general are a good fit, and another way to make this point crystal clear would be to define another enum such as:
```rust
enum FormatKind {
    Standard,
    NonStandard { name: String },
}
```

I believe that all of this also applies to people reading, reviewing or auditing the code later (including the author himself after a few months :stuck_out_tongue:). The information conveyed by `String`, `Option<String>` or `FormatKind` is obviously not the same and plays a role in readability.

How the value itself is actually decoded or encoded can remain an implementation detail in the body of the encoding and decoding functions, i.e.: if a single null terminator means that "no name is specified" and that is also means the format is "standard", then we can say that `Option::None` or `FormatKind::Standard` can be represented by this single null terminator.

I didn’t review everything yet though, so it’s possible I’m misunderstanding something. I hope it helps you to decide on that!

Reply:
Makes sense to use an empty string if it translates correctly to bytes.

Reply:
@CBenoit Well, in the case of `ClipboardFormat` PDU structure there is no option to specify absence of string, only an empty string (null terminator only) is possible, therefore I decided to reflect that in the Rust strucutre, But I agree that is it not idiomatic.

I just don't like the fact that if we go with Option<String> we will have the following situation when comparing encoded and decode:

```rust
let original = FormatList::new_unicode(&[ClipboardFormat { id: 42, name: Some("".to_string()) }], true);
let encoded = encode(&original);

let decoded = decode::<FormatList>(&encoded);

let actual = FormatList::new_unicode(&[ClipboardFormat { id: 42, name: None }], true)
assert_eq!(decoded, actual);

assert_eq!(decoded, original); // Will fail, re-encoding produces different results
```


I think the best idea would be to hide implementation details of `ClibpoardFormat` (make fields private) and automatically store empty strings as None.

```rust
pub struct ClipboardFormat {
    id: u32,
    name: Option<String>
}

impl ClipboardFormat  {
    pub fn new(id: u32) -> Self { ... }
    /// NOTE: Method does nothing if empty string is specified
    pub fn with_name(self, name: &str) -> Self { ... }

    pub fn id(&self) -> u32 { ... }
    pub fn name(&self) -> Option<&str> {...}
}
```

Reply:
Oh, I see. Sounds good to me!

Reply:
Note: since this is an implementation detail, you could indeed use `String` internally and simply check for emptiness when `name()` is called.
At your preference.